test/unit/org/apache/cassandra/security/DefaultSslContextFactoryTest.java

```diff
@@ -154,6 +162,54 @@ public void buildKeyManagerFactoryHappyPath() throws IOException
         Assert.assertTrue(defaultSslContextFactoryImpl3.checkedExpiry);
     }
 
+    @Test(expected = IOException.class)
+    public void buildOutboundKeyManagerFactoryWithInvalidKeystoreFile() throws IOException
+    {
+        Map<String, Object> config = new HashMap<>();
+        config.putAll(commonConfig);
+        config.put("outbound_keystore", "/this/is/probably/not/a/file/on/your/test/machine");
+
+        DefaultSslContextFactory defaultSslContextFactoryImpl = new DefaultSslContextFactory(config);
+        defaultSslContextFactoryImpl.checkedExpiry = false;
+        defaultSslContextFactoryImpl.buildKeyManagerFactory();
+    }
+
+    @Test(expected = IOException.class)
+    public void buildOutboundKeyManagerFactoryWithBadPassword() throws IOException
+    {
+        Map<String, Object> config = new HashMap<>();
+        config.putAll(commonConfig);
+        addOutboundKeystoreOptions(config);
+        config.put("outbound_keystore_password", "HomeOfBadPasswords");
+
+        DefaultSslContextFactory defaultSslContextFactoryImpl = new DefaultSslContextFactory(config);
+        defaultSslContextFactoryImpl.buildKeyManagerFactory();
+    }
+
+    @Test
+    public void buildOutboundKeyManagerFactoryHappyPath() throws IOException
```

Comment:
With these new tests we are testing outbound keymanager in isolation with inbound. However from the SslContextFactory class's perspective both (inbound and outbound) keystores exists together. Do you think it would be worth to add a test with both together? (Those tests can test for separate cert expiry checks etc).

Reply:
Added a check in the test case to check other flags. This will test the both the flags together.